### PR TITLE
Signing fixes back-ported from 2.x

### LINKED ISSUE #1998: Incorrect error from invalid private key for signTransaction
Hey guys,

I was running into an issue where I was getting an insufficient funds error when trying to sign a transaction. I found that this is a fairly common issue on the forums and it can be caused by having a private key that doesn't start with a 0x.

There is a simple solution to this in the packages/web3-eth-accounts/src/index.js file on line 145:
```
if (!privateKey) {
    error = new Error('No private key given!');

    callback(error);
    return Promise.reject(error);
}

if (!privateKey.startsWith('0x')) {
    error = new Error('Private key is invalid!');

    callback(error);
    return Promise.reject(error);
}
```
https://github.com/ethereum/web3.js/blob/v1.0.0-beta.36/packages/web3-eth-accounts/src/index.js#L145

I tested this from the v1.0.0-beta.36 tag and all tests passed.

### PR DESCRIPTION
**Pull request by nivida**
Fixes #1998, #2033, and #1074

### CODE REVIEW

File: packages/web3-eth-accounts/src/index.js [hunk 1]
```diff
@@ -168,36 +154,28 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
...
 
-            var rawTx = RLP.decode(rlpEncoded).slice(0, 6).concat(Account.decodeSignature(signature));
+            var validationResult = ethTx.validate(true);
 
-            rawTx[6] = makeEven(trimLeadingZero(rawTx[6]));
-            rawTx[7] = makeEven(trimLeadingZero(rawTx[7]));
-            rawTx[8] = makeEven(trimLeadingZero(rawTx[8]));
+            if (validationResult !== '') {
+                throw new Error('Signer Error: ' + validationResult);
+            }
```
Comment (cgewecke) [inline packages/web3-eth-accounts/src/index.js:168]: What do you think about adding the example tx in #1074 to the unit tests to see if it hits this validation check? 

It looks like this:
```js
signTxParams = { 
    data: "0x23123123123123123",  
    gas: 42424242424242,
    gasPrice: 42424242,
    nonce: 1,
    to: "0xd8b0990c007ba1ad97b37c001d1f87044312162a",
    value: "42424242",
    chainId: 1
}
```

File: packages/web3-eth-accounts/src/index.js [hunk 2]
```diff
@@ -168,36 +154,28 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
...
 
-            var rawTransaction = RLP.encode(rawTx);
+            var rlpEncoded = ethTx.serialize().toString('hex');
+            var rawTransaction = '0x' + rlpEncoded;
+            var transactionHash = utils.keccak256(rawTransaction);
```
Comment (cgewecke) [inline packages/web3-eth-accounts/src/index.js:172]: Is this transactionHash line extra?

**Review by cgewecke — APPROVED** [review]
Looks good! Just left a couple small notes

File: packages/web3-eth-accounts/src/index.js (hunk 2 above)
Comment (nivida) [reply packages/web3-eth-accounts/src/index.js:172]: I've added the ``transactionHash`` property to the returned object.

**Review by cgewecke — APPROVED** [review]
LGTM.

**Comment by cgewecke** [comment]
@nivida Apologies, I missed something important in the review here. Additionally, there was a bug in the E2E CI (fixed with #3138) which meant it wasn't obvious an eth.accounts.sign test was failing for geth dev.

The problem is that ethereumjs-tx requires [an additional object passed to the Transaction constructor][1] when a tx is being sent to a non-mainnet chain. You can see an example in the [ejs-tx signing test][2] removed from #3122.

[1]: https://github.com/ethereumjs/ethereumjs-tx#chain-and-hardfork-support
[2]:https://github.com/cgewecke/web3.js/blob/aa96ee8116fe1733cfbe60a61d9ca2ee9020b77a/test/e2e.method.signing.js#L31-L68

**Pull request merged**